### PR TITLE
Add workflow settings, typescript from Transport proejct

### PR DESCRIPTION
**Pull request by a-chabin**
I copied workflow settings from [Transport Repo](https://github.com/ekaterinburgdev/transport) by @hiba9201 and fix Vercel deploy.

- ESLint
- Prettier
- Typescript
- ~~Types for fetch~~

### CODE REVIEW

File: api/utils/fetch.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { HttpMethod, FetchOptions } from 'common/types/requests';
```
Comment (hiba9201) [inline api/utils/fetch.ts:1]: Not sure that this file is needed, because i wrote it for external ettu API (so there is CORS wrapper that we probably don't need for map). And maybe there will be other API response format

File: common/constants/colors.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const TRAM_COLOR = '#EC6608';
```
Comment (hiba9201) [inline common/constants/colors.ts:1]: ❓ Why did you move transport constants here? Let's leave only `coords.ts`

File: components/Map/Location/MapLocation.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        return () => {
+            map.stopLocate();
+        };
+    }, [map]);
+
+    return (
+        <Marker position={position || COORDS_EKATERINBURG}>
+            <Popup>Your location</Popup>
+        </Marker>
+    );
```
Comment (hiba9201) [inline components/Map/Location/MapLocation.tsx:35]: Can you do `return null` here? I don't think that we need marker of user on this map

File: next.config.js [hunk 4]
```diff
@@ -1,7 +1,10 @@
 /** @type {import('next').NextConfig} */
 const nextConfig = {
-  reactStrictMode: true,
-  swcMinify: true,
+    reactStrictMode: true,
+    swcMinify: true,
+    eslint: {
+        dirs: ['pages', 'components', 'api']
```
Comment (hiba9201) [inline next.config.js:6]: We should add `common` here too. I forgot to do it in my PR

File: api/utils/fetch.ts (hunk 1 above)
Comment (a-chabin) [reply api/utils/fetch.ts:1]: Yep, we already have our own API without problems with CORS. Remove it

File: common/constants/colors.ts (hunk 2 above)
Comment (a-chabin) [reply common/constants/colors.ts:1]: Thanks)

Remove it

File: components/Map/Location/MapLocation.tsx (hunk 3 above)
Comment (a-chabin) [reply components/Map/Location/MapLocation.tsx:35]: Done

File: next.config.js (hunk 4 above)
Comment (a-chabin) [reply next.config.js]: Done :)

File: components/Map/Location/MapLocation.tsx (hunk 3 above)
Comment (a-chabin) [reply components/Map/Location/MapLocation.tsx]: + remove unused imports

**Review by hiba9201 — APPROVED** [review]
